### PR TITLE
[Major] Dataloader: Just-In-Time tabularization

### PR DESCRIPTION
**Pull request by ourownstory**
Complete re-write of dataloader: Instead of tabularizing samples ahead of batching, samples are tabularized and features computed during batch creation. Trades Memory use for Compute.

This is an early implementation that works, but that needs to be compute-optimized.

- Addresses high data consumption when loading global models with many IDs and large datasets. See https://github.com/ourownstory/neural_prophet/issues/1469

### CODE REVIEW

File: tests/test_unit.py [hunk 1]
```diff
@@ -932,19 +979,18 @@ def test_handle_negative_values_replace():
 
 
 def test_add_country_holiday_multiple_calls_warning(caplog):
```
Comment (ourownstory) [inline tests/test_unit.py:981]: this could be moved to test_event_utils

File: tests/test_hdays_utils.py [hunk 2]
```diff
@@ -1,30 +1,23 @@
 #!/usr/bin/env python3
 
 import holidays
-import pytest
-
-from neuralprophet import hdays_utils
+from holidays import country_holidays
 
 
 def test_get_country_holidays():
```
Comment (ourownstory) [inline tests/test_hdays_utils.py:7]: this should be moved to test_event_utils

File: tests/test_hdays_utils.py [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
-        us_holidays = hdays_utils.get_country_holidays(country, years=2019)
+        us_holidays = country_holidays(country=country, years=2019)
         assert issubclass(us_holidays.__class__, holidays.countries.united_states.UnitedStates) is True
         assert len(us_holidays) == 10
 
-    with pytest.raises(AttributeError):
-        hdays_utils.get_country_holidays("NotSupportedCountry")
-
 
 def test_get_country_holidays_with_subdivisions():
```
Comment (ourownstory) [inline tests/test_hdays_utils.py:14]: this should be moved to test_event_utils

File: tests/test_integration.py [hunk 4]
```diff
@@ -473,12 +473,10 @@ def test_events():
         ["superbowl", "playoff"], lower_window=-1, upper_window=1, mode="multiplicative", regularization=0.5
     )
     # add the country specific holidays
-    m = m.add_country_holidays("US", mode="additive", regularization=0.5)
-    m.add_country_holidays("Indonesia")
+    m = m.add_country_holidays(
```
Comment (ourownstory) [inline tests/test_integration.py:476]: this whole test function should be moved to test_event_utils

File: tests/test_unit.py [hunk 5]
```diff
@@ -932,19 +979,18 @@ def test_handle_negative_values_replace():
...
     assert error_message not in caplog.text
 
-    m.add_country_holidays("Germany")
-    assert error_message in caplog.text
+    with pytest.raises(AssertionError):
+        m.add_country_holidays("Germany")
+        # assert error_message in caplog.text
 
 
 def test_multiple_countries():
```
Comment (ourownstory) [inline tests/test_unit.py:996]: this should be moved to test_event_utils

**Review by ourownstory — COMMENTED** [review]
Let's consolidate our tests for holidays and events!

**Review by MaiBe-ctrl — APPROVED** [review]
Good job!

**Pull request merged**